### ISSUE TITLE
Query 10 rows of data before and after the keyword

### ISSUE THREAD

**Issue by ly0511** [VictoriaMetrics/VictoriaMetrics#5007]
### Describe the question in detail

If I want to query n rows of data before and after a certain keyword, for example, if I want to view the first 10 rows and the last 10 rows of all rows of panic, how should I write the query statement

Currently, I can only query data for the row where panic resides using AND panic

![image](https://github.com/VictoriaMetrics/VictoriaMetrics/assets/23255488/a23aebe5-04f4-4c1a-b31f-e4e2e909ce30)

### Troubleshooting docs

- [ ] General - https://docs.victoriametrics.com/Troubleshooting.html
- [ ] vmagent - https://docs.victoriametrics.com/vmagent.html#troubleshooting
- [ ] vmalert - https://docs.victoriametrics.com/vmalert.html#troubleshooting

**Comment by zekker6** [comment]
@ly0511 Currently this is not possible, so I've converted this into a feature request.

**Comment by ly0511** [comment]
@zekker6 Whether the limit of 1000 rows in the query result can be removed and all data in the log can be returned.
If we can return all the data, I think we can solve the above problem

**Comment by zekker6** [comment]
@valyala @Loori-R What do you think?
Technically we can allow displaying larger amount of logs, but maybe we should use some kid of pagination instead of just increasing amount of logs?

**Comment by Loori-R** [comment]
You're right. While the frontend can indeed lift the limitation and visually paginate, loading large volumes of data at once might result in performance issues. 
Implementing pagination on the backend is a more efficient approach.

**Comment by ly0511** [comment]
> @ly0511 Currently this is not possible, so I've converted this into a feature request.
@zekker6 I want to know how long it will take for the function to be supported？

**Comment by zekker6** [comment]
@ly0511 There is no ETA yet, please keep track of the roadmap [here](https://docs.victoriametrics.com/VictoriaLogs/Roadmap.html) and new releases changelog [here](https://docs.victoriametrics.com/VictoriaLogs/CHANGELOG.html).
